apps/agentic-chat/src/components/assistant-ui/BebopQuoteUI.tsx

```diff
@@ -11,101 +14,87 @@ import { Input } from '../ui/input';
 import { AlertCircle, ArrowRightLeft } from 'lucide-react';
 import { TextShimmer } from '../TextShimmer';
 import { Button } from '../ui/button';
+import { useAssetsStore } from '../../stores/assets';
+import { BebopRateParams, BebopRateResult } from '../../tools/bebopRate';
 
-// Types for bebopRate tool args and result
-export type BebopRateArgs = {
-  chain: string;
-  fromAsset: {
-    address: string;
-    decimals: number;
-    symbol?: string;
-  };
-  toAsset: {
-    address: string;
-    decimals: number;
-    symbol: string;
-  };
-  sellAmountCryptoPrecision: string;
-  fromAddress: string;
-};
+const BebopUiContent: ToolCallContentPartComponent<
+  BebopRateParams,
+  BebopRateResult
+> = ({ status, result, args, isError }) => {
+  const assetsStore = useAssetsStore();
+  const sellAsset = assetsStore.assetsById[args.sellAssetId];
+  const buyAsset = assetsStore.assetsById[args.buyAssetId];
 
-export type BebopRateResult = {
-  sellAmountCryptoPrecision: string;
-  buyAmountCryptoPrecision: string;
-  sellAsset: {
-    symbol: string;
-  };
-  buyAsset: {
-    symbol: string;
-  };
-  approvalTarget: string;
-};
+  switch (status.type) {
+    case 'running':
+    case 'requires-action':
+    case 'incomplete': {
```

Comment:
Missing states explicit handling + moved to a `Content` component so we can leverage zustand hook + added a state for args not yet here. Everything else stays the same.